advanced-python/4bHyperparameterTuning.ipynb

```diff
@@ -0,0 +1,420 @@
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def plot_mass(df):\n",
+    "    counts, bins, _ = plt.hist(df['Jpsi_M'], bins=100, range=[2.75, 3.5], histtype='step')\n",
+    "    # You can also use LaTeX in the axis label\n",
+    "    plt.xlabel('$J/\\\\psi$ mass [GeV]')\n",
+    "    plt.xlim(bins[0], bins[-1])"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 3,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def plot_comparision(var, mc_df, bkg_df):\n",
+    "    _, bins, _ = plt.hist(mc_df[var], bins=100, histtype='step', label='MC', density=1)\n",
+    "    _, bins, _ = plt.hist(bkg_df[var], bins=bins, histtype='step', label='Background', density=1)\n",
+    "    plt.xlabel(var)\n",
+    "    plt.xlim(bins[0], bins[-1])\n",
+    "    plt.legend(loc='best')"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 4,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def plot_roc(bdt, training_data, training_columns, label=None):\n",
+    "    y_score = bdt.predict_proba(training_data[training_columns])[:,1]\n",
+    "    fpr, tpr, thresholds = roc_curve(training_data['catagory'], y_score)\n",
+    "    area = auc(fpr, tpr)\n",
+    "\n",
+    "    plt.plot([0, 1], [0, 1], color='grey', linestyle='--')\n",
+    "    if label:\n",
+    "        plt.plot(fpr, tpr, label=f'{label} (area = {area:.2f})')\n",
+    "    else:\n",
+    "        plt.plot(fpr, tpr, label=f'ROC curve (area = {area:.2f})')\n",
+    "    plt.xlim(0.0, 1.0)\n",
+    "    plt.ylim(0.0, 1.0)\n",
+    "    plt.xlabel('False Positive Rate')\n",
+    "    plt.ylabel('True Positive Rate')\n",
+    "    plt.legend(loc='lower right')\n",
+    "    # We can make the plot look nicer by forcing the grid to be square\n",
+    "    plt.gca().set_aspect('equal', adjustable='box')"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 5,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def plot_significance(bdt, training_data, training_columns, label=None):\n",
+    "    y_score = bdt.predict_proba(training_data[training_columns])[:,1]\n",
+    "    fpr, tpr, thresholds = roc_curve(training_data['catagory'], y_score)\n",
+    "\n",
+    "    n_sig = 1200\n",
+    "    n_bkg = 23000\n",
+    "    S = n_sig*tpr\n",
+    "    B = n_bkg*fpr\n",
+    "    metric = S/np.sqrt(S+B)\n",
+    "\n",
+    "    plt.plot(thresholds, metric, label=label)\n",
+    "    plt.xlabel('BDT cut value')\n",
+    "    plt.ylabel('$\\\\frac{S}{\\\\sqrt{S+B}}$')\n",
+    "    plt.xlim(0, 1.0)\n",
+    "\n",
+    "    optimal_cut = thresholds[np.argmax(metric)]\n",
+    "    plt.axvline(optimal_cut, color='black', linestyle='--')"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 6,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "#max_entries = 1000\n",
+    "data_df = uproot.open('/eos/user/l/lhcbsk/advanced-python/data/real_data.root')['DecayTree'].pandas.df()#entrystop=max_entries)\n",
```

Comment:
We shouldn't rely on the EOS mount. Those files are available with HTTPS:
* `https://cern.ch/starterkit/data/advanced-python-2018/real_data.root`
* `https://cern.ch/starterkit/data/advanced-python-2018/simulated_data.root`

Reply:
ah okay, I'd just thought I'd forgotten 'root://eosuser.cern.ch/', thanks!

Reply:
I'd rather use `root://` to be more realistic but unfortunately it's not possible to make parts of `root://eosuser.cern.ch/` completely public.